src/cicd/actions.ts

```diff
@@ -0,0 +1,245 @@
+import { Repository } from 'aws-cdk-lib/aws-codecommit';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import {
+  CodeBuildStep,
+  CodePipelineSource,
+  ConnectionSourceOptions,
+  IFileSetProducer,
+  ShellStep,
+} from 'aws-cdk-lib/pipelines';
+import { Construct } from 'constructs';
+import { getCodeartifactPublishPolicyStatements } from './utils';
+
+export interface GetSynthActionProps {
+  readonly codePipelineSource?: IFileSetProducer;
+  readonly cdkVersion?: string;
+  readonly partition?: string;
+  readonly region?: string;
+  readonly account?: string;
+  readonly rolePolicyStatements?: PolicyStatement[];
+  readonly codeartifactRepository?: string;
+  readonly codeartifactDomain?: string;
+  readonly codeartifactDomainOwner?: string;
+}
+
+export interface CodeCommitSourceActionProps {
+  readonly scope: Construct;
+  readonly repositoryName: string;
+  readonly branch: string;
+  readonly props?: ConnectionSourceOptions;
+}
+
+export function getCodeCommitSourceAction(props: CodeCommitSourceActionProps) {
```

Comment:
It would be good to annotate the return value of this function as well. Like:
```typescript
export function getCodeCommitSourceAction(props: CodeCommitSourceActionProps): CodePipelineSource {}
```